source/develop/release-management/features/storage/pass_discard_from_guest_to_underlying_storage.html.md

```diff
@@ -0,0 +1,234 @@
+---
+authors: ishaby
+wiki_title: Features/PassDiscardFromGuestToUnderlyingStorage
+feature_name: Pass discard from guest to underlying storage
+feature_modules: engine/vdsm/api-model
+feature_status: Development
+---
+
+# Pass discard from guest to underlying storage
+
+## Summary
+
+### How we use thin provisioned disks today
+Today, when we create a thin provisioned disk on a block domain, a logical volume with a size of 1G is created, regardless its virtual size.<br/>
+When the free space is about to end, we extend the lv with one more Gigabyte, so that the disk grows again and again until it reaches its virtual size.<br/>
+If the lun that is used to create the storage domain is also thin provisioned, it behaves in a similar way. That is, when we extend our disk with one Gigabyte, we don't necessarily get 1G from the storage, but we do get 1G of virtual space.
+
+Similarly, in file domains the underlying thin provisioned lun gets bigger as the corresponding disk's file grows.
+
+### What we would like to achieve
+When files are removed from the disk by the guest, the lv does not shrink (it would if we called `lvreduce`, but we don't call it for technical reasons). Since no change was made to the size of the lv, the thin provisioned lun in the underlying storage haven't changed either.
+So in fact, although we freed some of the disk's space, the underlying storage is not aware of it, and practically we got a situation where the underlying thin provisioned luns can only grow, and can never reduce.
+
+The same goes with file storage - when files are removed by the guest, the disk's actual size remains the same and so is the size of the underlying thin provisioned luns.
+
+A better approach would be to "tell" the underlying storage that we freed some space that we don't need anymore, so that someone else can use it.
+
+### For whom is this feature useful
+This feature is pretty useful for anyone who uses thin provisioned disks and wants to utilize them more efficiently.
```

Comment:
Remove the fluff: Users that have thinly-provisioned LUNs and wish to reclaim space released by their guests.